Updates and refactoring of YAMLScript 0.1.82

ingydotnet opened the pull request:
This was pinned to 0.1.56, now 0.1.82.

We take advantage of new additions to make the inputs even cleaner.

Generated `config.yml` file remains unchanged (as required for any refactoring).

ingydotnet committed:
Updates and refactoring of YAMLScript 0.1.82

This was pinned to 0.1.56, now 0.1.82.

We take advantage of new additions to make the inputs even cleaner.

ingydotnet commented:
This work uncovered a bug in 0.1.81 which has been fix, but not released. 
Merging this should not be done until 0.1.82 is released. Scheduled for 2024/11/03.

ingydotnet commented:
YAMLScript 0.1.82 has been released, and tested to work with this PR.

Ready for review.

PEZ commented on .circleci/workflows/build-test.yaml:
```diff
@@ -1,44 +1,30 @@
-- test-cljslib:
-    requires:
-    - build
-
-- test-integration:
-    requires:
-    - build
+- prettier-check:: requires-checkout
+- test-grammar::   requires-checkout
+- build::          requires-checkout
 
-- test-e2e:
-    requires:
-    - build
-
-- test-e2e-sub-projects:
-    requires:
-    - build
-
-- test-ts-unit:
-    requires:
-    - build
+- eslint-check::          requires-build
+- test-cljslib::          requires-build
+- test-integration::      requires-build
+- test-e2e::              requires-build
+- test-e2e-sub-projects:: requires-build
+- test-ts-unit::          requires-build
```
This is sweet! Just a question: Would it be possible to make the required job an argument somehow? Not saying we should do it, just wondering.

ingydotnet replied:
https://gist.github.com/ingydotnet/1f8b535cc3e157a6835ec21406c0914d#file-gistfile1-txt-L32-L34
shows 2 possible ways to do it differently (both keeping definitions to single lines).

You could predefine a `requires` fn that takes an arg,
or you could predefine nothing and just use YAML flow syntax.

Might be best to just stick with YAML syntax here as nothing very interesting is happening that really requires a function.

ingydotnet replied:
See 2082ad149

PEZ commented on .circleci/workflows/release-publish.yaml:
```diff
@@ -1,91 +1,59 @@
+!yamlscript/v0/
+
+# Define 2 default mappings for clean reuse:
+default-filters =::
+  branches:
+    ignore: /.*/
+  tags:
+    only: /^v\d+\.\d+\.\d+-?.*/
+
+# XXX This might not be required to differ from default-filters
+default-filters-2 =::
+  branches:
+    ignore: /.*/
+  tags:
+    only: /^v\d+\.\d+\.\d+$/
```
I think the names of these filters could be `prerelease-filter` and `release-filter`.

ingydotnet replied:
See f694be9e2
I made comments on the prerelease regex.
I won't change the regex for this PR, since it's a refactoring and should cause no normative changes.

PEZ replied:
Yeah, I think I may have just wanted to hint at the intention. Seems right to make it actually stop malformed release tags like you suggest. But later, of course.

The naming could probably be clearer still. Like `include-prerelease-tags` and `only-release-tags`, or something. But we can do that later as well.

PEZ reviewed (commented):
LGTM. I only had one small comment on the naming of the tag filters.

ingydotnet committed:
Just use YAML flow collections instead of special vars

ingydotnet committed:
Define an inline filters fn to reduce repetition

I like this regex is not great:
/^v\d+\.\d+\.\d+-?.*/

It's no different than:
/^v\d+\.\d+\.\d+.*/

Perhaps this was intended or is preferred:
/^v\d+\.\d+\.\d+(?:-\S+)?$/

PEZ commented:
Thanks for doing this! 🙏 ❤️
